src/transformers/models/whisper/tokenization_whisper.py

```diff
@@ -838,17 +840,34 @@ def get_prompt_ids(self, text: str, return_tensors="np"):
         batch_encoding.convert_to_tensors(tensor_type=return_tensors)
         return batch_encoding["input_ids"]
 
-    @staticmethod
-    def _strip_prompt(token_ids: List[int], prompt_token_id: int, decoder_start_token_id: int):
-        has_prompt = isinstance(token_ids, list) and token_ids and token_ids[0] == prompt_token_id
+    def _strip_prompt(
+        self,
+        token_ids: Union[List[int], np.ndarray, torch.Tensor, tf.Tensor],
+        prompt_token_id: int,
+        decoder_start_token_id: int,
+    ):
+        has_prompt = (
+            isinstance(token_ids, (List[int], np.ndarray, torch.Tensor, tf.Tensor))
+            and token_ids
+            and token_ids[0] == prompt_token_id
+        )
         if has_prompt:
+            if not isinstance(token_ids, list):
+                token_ids = self._convert_to_list(token_ids)
             if decoder_start_token_id in token_ids:
                 return token_ids[token_ids.index(decoder_start_token_id) :]
             else:
                 return []
 
         return token_ids
 
+    @staticmethod
+    def _convert_to_list(token_ids):
+        if isinstance(token_ids, (torch.Tensor, np.ndarray)):
```

Comment:
What you can do here is check whether `token_ids` has the method `numpy`, and subsequently convert it to a numpy array:
```python
if hasattr(token_ids, "numpy"):
    token_ids = token_ids.numpy()
# now the token ids are either a numpy array, or a list of lists
if isinstance(token_ids, np.ndarray):
    token_ids = token_ids.tolist()
# now the token ids are always a list of lists
return token_ids
```

It might be more robust checking that we perform this on Torch/TF tensors:
```python
if "torch" in str(type(token_ids)) or "tensorflow" in str(type(token_ids)) and hasattr(token_ids, "numpy"):
    token_ids = token_ids.numpy()
# now the token ids are either a numpy array, or a list of lists
if isinstance(token_ids, np.ndarray):
    token_ids = token_ids.tolist()
# now the token ids are always a list of lists
return token_ids
```

Reply:
the first option is more elegant than the second but I decided to take the second.